### ISSUE TITLE
SC-IDE: Icon for unsaved document is different in the tab bar and documents docklet

### ISSUE THREAD

**Issue by kisielk** [supercollider/supercollider#3054]
In the documents docklet it's a floppy disk, in the tab bar it's a hard drive.

![screen shot 2017-07-18 at 23 46 46](https://user-images.githubusercontent.com/58618/28354119-63107ab8-6c13-11e7-86f9-d26b488166da.png)

**Comment by telephon** [comment]
is there an alternative standard symbol in QT? both harddrive and floppy are losing their meaning more and more.

**Comment by brianlheim** [comment]
@telephon 

Here are some images I found of all the QT-provided symbols:
http://cdn-ak.f.st-hatena.com/images/fotolife/n/nukesaq88/20130412/20130412010439.png
http://cdn-ak.f.st-hatena.com/images/fotolife/n/nukesaq88/20130412/20130412010429.png

I think the popular non-floppy alternative seems to be a down arrow pointing to some abstract rectangle, or a down arrow pointing to a hard drive:
https://www.google.com/search?q=save+icon&safe=off&source=lnms&tbm=isch&sa=X&ved=0ahUKEwjZj5mi75fVAhVEWD4KHUOeCHwQ_AUICigB&biw=1092&bih=774&dpr=2

Qt doesn't seem to have a standard icon that looks like that. I would love to switch over to a new icon set at some point, because in general the ones provided by Qt feel pretty outdated, but that is a separate issue from this one.

@kisielk 

`QIcon::fromTheme("document-save")` seems to be problematic on Windows, because no icon actually shows up. So that's out.

On Windows, here's what the editor looks like with the hard drive icon in the editor:

![capture](https://user-images.githubusercontent.com/15369640/28418399-8bece3ec-6d29-11e7-892d-a7e6bbdbfd3e.PNG)

and with the floppy:

![capture2](https://user-images.githubusercontent.com/15369640/28418401-8fb74d8c-6d29-11e7-94de-b21ecbe8cf49.PNG)

My preference is to use `SP_DialogSaveButton` (floppy) on Windows/Linux and `SP_DriveHIcon` (HDD) on macOS. That drive icon is still used in apple's GUIs, so I think it will be familiar to more users. Not a strong preference. I do however think we should resolve this inconsistency ASAP using the icons we have available.

**Comment by kisielk** [comment]
On macOS the standard is to append the text " - Edited" to the title bar, often in a lighter font. I can't find any application on my computer that uses an icon to indicate unsaved content. 

The closest is Visual Studio Code, which replaces the X for closing a tab with a dot when it has been edited, but that application doesn't really conform to the standard OS look and feel.

That being said, for the time being it's probably fine to just use an icon as long as it's consistent... I agree with your last paragraph.

**Comment by brianlheim** [comment]
I've never thought about this before! I did a quick survey:

Xcode: darken the icon when unsaved
Word: grey the icon when unsaved
Atom: x changes to dot like VS
Sublime: x changes to dot
Adobe products: add "*" to title bar (Illustrator, Photoshop), or append (— Edited) (Experience Design)
IntelliJ: blue the filename in the titlebar
iTerm: blue dot instead of X in a tab bar to indicate unread

I couldn't find anything that says the standard is ++ "— Edited". The only thing I could find is this: http://osxdaily.com/2011/05/25/the-window-titlebar-in-mac-os-x-tells-you-when-you-have-unsaved-file-changes/ which suggests darkening the icon is standard (but it is from 2011, so idk).

Regardless, I agree we can do better than adding an icon. These solutions are all OK with me!

**Comment by kisielk** [comment]
3rd recommendation here: https://developer.apple.com/macos/human-interface-guidelines/system-capabilities/auto-save/

**Comment by brianlheim** [comment]
Ah ok. I'm a little confused though:

> In general, use a dot to mark a modified document as unsaved only when it can't be autosaved. If a document can be autosaved, there shouldn’t be a dot on the close button in the document’s title bar and next to the document’s name in the Window menu. It’s fine to append an Edited suffix to the document's title in the title bar, but this suffix should be removed when the document is closed or the user manually performs a save.

From that it seems like the dot is more standard. We don't have autosaving in the IDE AFAIK (we have backups in the case of crashes, but I think autosaving would overwrite the original document, not create backups).

**Comment by kisielk** [comment]
Yes, in this case the dot is the one in the top left corner which normally closes the window. That doesn't apply here because the documents are represented by tabs and not individual windows.

**Comment by eleses** [comment]
On MS Windows, since SC 3.6.6. there isn't a "different" icon on the tabs. There just isn't one! It's certainly related to this, but more severe, so I'll probably file it as a different bug soon.
